java/client/src/main/java/glide/api/GlideClient.java

```diff
@@ -568,4 +574,72 @@ public CompletableFuture<Object[]> scan(
         GlideString[] arguments = new ArgsBuilder().add(cursor).add(options.toArgs()).toArray();
         return commandManager.submitNewCommand(Scan, arguments, this::handleArrayResponseBinary);
     }
+
+    @Override
+    public CompletableFuture<Object> invokeScript(@NonNull Script script) {
+        if (script.getBinaryOutput()) {
+            return commandManager.submitScript(
+                    script, List.of(), List.of(), this::handleBinaryObjectOrNullResponse);
+        } else {
+            return commandManager.submitScript(
+                    script, List.of(), List.of(), this::handleObjectOrNullResponse);
+        }
+    }
+
+    @Override
+    public CompletableFuture<Object> invokeScript(
+            @NonNull Script script, @NonNull List<String> keys, List<String> args) {
+        if (script.getBinaryOutput()) {
+            return commandManager.submitScript(
+                    script,
+                    keys.stream().map(GlideString::gs).collect(Collectors.toList()),
+                    args.stream().map(GlideString::gs).collect(Collectors.toList()),
+                    this::handleBinaryObjectOrNullResponse);
+        } else {
+            return commandManager.submitScript(
+                    script,
+                    keys.stream().map(GlideString::gs).collect(Collectors.toList()),
+                    args.stream().map(GlideString::gs).collect(Collectors.toList()),
+                    this::handleObjectOrNullResponse);
+        }
+    }
+
+    @Override
+    public CompletableFuture<Object> invokeScriptBinary(
+            @NonNull Script script, List<GlideString> keys, List<GlideString> args) {
+        if (script.getBinaryOutput()) {
```

Comment:
I think this should be assumed as always true for this API

Reply:
> Indication if script invocation output can return binary data.

The script can also return nothing no? or am I misunderstanding this attribute

Reply:
Right, but if user expects to receive `byte[]` instead of strings, this parameter set to true